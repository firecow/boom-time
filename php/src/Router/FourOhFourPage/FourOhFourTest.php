<?
declare(strict_types=1);

namespace App\Router\FourOhFourPage;

use App\Exceptions\FileErrorException;
use App\Exceptions\FileNotFoundException;
use App\Mocks\ContextMock;
use App\Mocks\SuperGlobalsMock;
use App\Testing\RegressionTest;

class FourOhFourTest extends RegressionTest
{
    /**
     * @throws FileErrorException
     * @throws FileNotFoundException
     */
    public function doTest(): string
    {
        $superGlobals = new SuperGlobalsMock();
        $ctx = new ContextMock($superGlobals);
        $fourOhFourResponse = FourOhFour::generateFourOhFourResponse($ctx);
        return $fourOhFourResponse->prettyPrint();
    }
}
