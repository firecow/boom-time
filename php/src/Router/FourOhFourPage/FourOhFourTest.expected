HTTP/1.1 404 Not Found
Content-Length: 372
Content-Type: text/html

<html>
<head>
    <meta name="viewport" content="initial-scale=1">
    <title>Oh Snap</title>
    <link rel="shortcut icon" href="/favicon.png" type="image/x-icon"/>
    <link rel="stylesheet" type="text/css" href="/style.css">
</head>
<body>

<div class="padding">
    <div>Oh Snap (404)</div>
    <div>The page you requested cannot be found</div>
</div>

</body>
</html>